File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/coverletter/CoverLetterApplicationDto.kt
package ch.streckeisen.mycv.backend.coverletter

data class CoverLetterApplicationDto(
    val jobTitle: String?,
    val company: String?,
    val contactPerson: CoverLetterContactPersonDto?,
    val addressee: String?,
    val salutation: String?,
    val companyAddress: CoverLetterCompanyAddressDto?,
    val content: String?,
    val closing: String?
)

data class CoverLetterCompanyAddressDto(
    val street: String?,
    val postcode: String?,
    val city: String?
)

data class CoverLetterContactPersonDto(
    val firstName: String?,
    val lastName: String?
)

File: frontend/src/types/coverletter/CoverLetterApplicationDto.ts
export interface CoverLetterApplicationDto {
  jobTitle: string | undefined
  company: string | undefined
  contactPerson: CoverLetterContactPersonDto | undefined
  addressee: string | undefined
  salutation: string | undefined
  companyAddress: CoverLetterCompanyAddressDto | undefined
  coverLetterContent: string | undefined
  closing: string | undefined
}

export interface CoverLetterCompanyAddressDto {
  street: string | undefined
  postcode: string | undefined
  city: string | undefined
}

export interface CoverLetterContactPersonDto {
  firstName: string | undefined
  lastName: string | undefined
}
